server/fleet/service.go

```diff
@@ -711,6 +711,11 @@ type Service interface {
 	// error can be raised to the user.
 	VerifyMDMWindowsConfigured(ctx context.Context) error
 
+	// VerifyMDMAppleOrWindowsConfigured verifies that the server is configured
+	// for either Apple or Windows MDM. If an error is returned, authorization is
+	// skipped so the error can be raised to the user.
+	VerifyMDMAppleOrWindowsConfigured(ctx context.Context) error
```

Comment:
@gillespi314 I went ahead and created this middleware that checks if either Apple or Windows MDM is enabled and fails only if both aren't. I applied it to your disk encryption summary endpoint.